crates/matrix-sdk-base/src/sliding_sync/mod.rs

```diff
@@ -698,14 +698,34 @@ async fn cache_latest_events(
     let mut encrypted_events =
         Vec::with_capacity(room.latest_encrypted_events.read().unwrap().capacity());
 
+    // Try to get room power levels from the current changes
```

Comment:
This feels like a lot of work to do just in case you get a knocked state event, which I presume doesn't happen very often.

Reply:
I agree, but I didn't find any alternatives other than not implementing this part of the functionality 🫤 . We'd have to change the sync process so we can process these latest events *after* the client has processed the sync request and saved all the data to the store and I think that would be quite complex.